Readme.md

```diff
@@ -1,40 +1,39 @@
 <!-- default badges end -->
-<!-- default file list -->
-*Files to look at*:
 
-* [ActionsInvokerViewController.cs](./CS/EFCore/ExecuteActionEF/ExecuteActionEF.Module/Controllers/ActionInvokerViewController.cs) 
-<!-- default file list end -->
-# How to execute Actions in code
+# XAF - How to execute Actions programmatically
 
-## Scenario
+This example explains how you can trigger the code inside `Execute` event handlers of custom or standard XAF Actions. For example, you can programmatically invoke an Action that shows a View or runs code in the background.
 
-Normally, XAF Actions are executed only by end users using the interaction with UI elements (for instance, button clicks, navigation, or main menu selection). In advanced UI scenarios, XAF developers may want to programmatically trigger the code inside the Execute event handlers of custom or standard XAF Actions. For instance, a XAF developer wants to execute default Action logic that shows a View or runs some code in the background.
+> **NOTE** 
+> We do not recommend this approach. Refactor your Action's event handlers and extract required code into separate methods. Call these separate methods directly without triggering UI-related Action code.
 
-In general, this is rarely needed and is not recommended because there are always better solutions. For instance, you may avoid this if you refactor your Action's Execute and other event handlers and extract the required code into separate methods (then, you can call these methods directly without triggering the UI-related Action code).
+![image](https://user-images.githubusercontent.com/14300209/227978022-a50434fa-413f-4634-b0ba-09936fd82927.png)
 
-## Solution
+## Implementation Details
 
 ### SimpleAction
 
 `mySimpleAction.DoExecute();`
 
 ### ParametrizedAction
 
-`myParametrizedAction.DoExecute("test value"); // Mind the ParametrizedAction.ValueType. Here, a string is used as an example.`
+`myParametrizedAction.DoExecute("test value");`
 
-### SingleChoiceAction
+Make sure that the parameter type matches `ParametrizedAction.ValueType`. A string is used in this example.
 
-`mySingleChoiceAction.DoExecute(mySingleChoiceAction.Items[0]); // Mind the SingleChoiceAction.Items structure. Here, the first item is used as an example.`
+### SingleChoiceAction
 
-### PopupWindowShowAction (to display a dialog window)
+`mySingleChoiceAction.DoExecute(mySingleChoiceAction.Items[0]);`
 
-We don't have a public API to execute a PopupWindowShowAction. Use the [ShowViewStrategyBase.ShowViewInPopupWindow(View, Action, Action, String, String)](https://docs.devexpress.com/eXpressAppFramework/DevExpress.ExpressApp.ShowViewStrategyBase.ShowViewInPopupWindow(DevExpress.ExpressApp.View-System.Action-System.Action-System.String-System.String)) method to show a view in a popup.
+Make sure the pass a parameter that's a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection.
```

Comment:
Make sure you pass a parameter that is a valid item from `SingleChoiceAction.Items`. This example passes the first item in the collection